src/tools/grpc_server.cc

```diff
@@ -243,6 +246,40 @@ static std::string pat(const std::shared_ptr<RE2> &p) {
     return p->pattern();
 }
 
+void CodeSearchImpl::TagsFirstSearch_(::CodeSearchResult* response, query& q, string& line_pat, match_stats& stats) {
```

Comment:
Let's make this a `const string &` to make it clear we're not mutating. I might even just pass in `q` and extract it again via `q.line_pat->pattern()` to drop a parameter.

Reply:
I'll make it `const` right now!

As for calling `pattern()` again — I didn't see any indication of whether `pattern()` is expensive or not (I didn't check whether it has to do something like rebuild the RE's string from an introspection of its bytecode or something), hence my temptation to go ahead and pass the value while I had it already. Is it actually a cheap operation?

Reply:
https://github.com/google/re2/blob/master/re2/re2.h#L264

it's stored directly internally, which is what I'd expect -- the bytecode is going to be a lossy translation, since it canonicalizes, and I'd expect `pattern()` to return the input unmodified.

Reply:
Thanks for tracking down the line number! (I hadn't had time to yet today.) I'll go remove that parameter —